scalanet/src/io/iohk/scalanet/peergroup/dynamictls/DynamicTLSPeerGroup.scala

```diff
@@ -141,6 +142,102 @@ object DynamicTLSPeerGroup {
     }
   }
 
+  final case class ConfigError(description: String)
+
+  sealed abstract case class FramingConfig private (
+      maxFrameLength: Int,
+      lengthFieldOffset: Int,
+      lengthFieldLength: Int,
+      encodingLengthAdjustment: Int,
+      decodingLengthAdjustment: Int,
+      initialBytesToStrip: Int,
+      failFast: Boolean,
+      byteOrder: ByteOrder,
+      lengthIncludesLengthFieldLength: Boolean
+  )
+
+  object FramingConfig {
+    private val possibleLengthFieldLength = Set(1, 2, 3, 4, 8)
+
+    /**
+      *
+      * Configures framing format for all the peers in PeerGroup
+      *
+      * Check [[io.netty.handler.codec.LengthFieldPrepender]] and [[io.netty.handler.codec.LengthFieldBasedFrameDecoder]]
+      * for good description of all the fields
+      *
+      * @param maxFrameLength the maximum length of the frame. If the length of the frame is greater than this value
+      *                       channel with generate DecodingError event.
+      * @param lengthFieldOffset the offset of the length field.
+      * @param lengthFieldLength  the length of the length field. Must be 1, 2, 3, 4 or 8 bytes.
+      * @param decodingLengthAdjustment  the compensation value to add to the value of the length field when decoding.
+      * @param encodingLengthAdjustment the compensation value to add to the value of the length field when encoding.
+      * @param initialBytesToStrip the number of first bytes to strip out from the decoded frame. In standard framing setup
+      *                            it should be equal to lengthFieldLength.
+      * @param failFast  If true, a DecodingError event is generated as soon as the decoder notices the length of the frame will
+      *                  exceed maxFrameLength regardless of whether the entire frame has been read. If false,
+      *                  a DecodingError event is generated after the entire frame that exceeds maxFrameLength has been read.
+      * @param byteOrder the ByteOrder of the length field.
+      * @param lengthIncludesLengthFieldLength if true, the length of the prepended length field is added to the value of the prepended length field.
+      */
+    def buildConfig(
+        maxFrameLength: Int,
+        lengthFieldOffset: Int,
+        lengthFieldLength: Int,
+        initialBytesToStrip: Int,
+        encodingLengthAdjustment: Int = 0,
+        decodingLengthAdjustment: Int = 0,
+        failFast: Boolean = true,
+        byteOrder: ByteOrder = ByteOrder.BIG_ENDIAN,
+        lengthIncludesLengthFieldLength: Boolean = false
+    ): Either[ConfigError, FramingConfig] = {
+      val smallEnoughOffset = lengthFieldOffset <= maxFrameLength - lengthFieldLength
+      for {
+        _ <- Either.cond(maxFrameLength > 0, (), ConfigError("maxFrameLength should be positive"))
```

Comment:
A small helper method could turn these into something like `check(maxFrameLength > 0, "maxFrameLength should be positive")`